probably/pgcl/parser.py

```diff
@@ -441,7 +441,8 @@ def _parse_query(t: Tree):
                 assert isinstance(lit, RealLitExpr)
                 return PlotInstr(VarExpr(_parse_var(_child_tree(t, 0))),
                                  prob=lit)
-            elif t.children[1].data in ('true', 'false'):
+            elif t.children[1].data in ('true', 'false') or \
+                    (t.children[1].data == 'var' and t.children[1].children[0] in ('true', 'false')):
                 raise SyntaxError("Plot instruction does not support boolean operators")
```

Comment:
This is a workaround for the problem mentioned [here](https://github.com/Philipp15b/probably/pull/3/commits/f1d923a17ae6e479ee14665c8610c88c5e4eb1b7#r888889634), so we get a nicer error message. This check should still be moved to the typechecker, but I don't see how this can be done without some extensive rewrites, as the parsing of a `PlotInstr` already depends on the type of the literal.